File: UI_testing/test_node_selenium/page_objects/sonarqube_page.js
const BaseElement = require('./base_element')
const constants = require('../constants')

class SonarqubePage extends BaseElement {
    constructor(driver, _css, _xpath) {
        super(driver, _css, _xpath)
        this.projectsTab = new BaseElement(driver, '', constants.PROJECTS_TAB_XPATH)
        this.pageOverview = new BaseElement(driver, constants.PROJECT_OVERVIEW_SELECTOR)
    }

    async clickProjectsTab() {
        await this.projectsTab.click()
    }

    async clickProject(project) {
        this.projectLink = new BaseElement(this.driver, '', `//h2[@class="project-card-name"]/a[text()="${project}"]`)
        this.projectLink.click()
    }

    async screenshotProject() {
        await this.pageOverview.screenshot()
    }
}

module.exports = SonarqubePage

File: UI_testing/test_python_selenium/page_objects/sonarqube_page.py
from .base_element import BaseElement
import constants

class SonarqubePage(BaseElement):
    def __init__(self, driver, _css='', _xpath=''):
        super().__init__(driver, _css, _xpath)
        self.portfolio_tab = BaseElement(self.driver, '', constants.PROJECTS_TAB_XPATH)
        self.overview_page = BaseElement(self.driver, constants.PROJECT_OVERVIEW_SELECTOR)

    def click_portfolio_tab(self):
        self.portfolio_tab.click()

    def click_project(self, name):
        self.project_link = BaseElement(self.driver, '', '//h2[@class="project-card-name"]/a[text()="{}"]'.format(name))
        self.project_link.click()

    def screenshot_project(self, img_name):
        self.overview_page.screenshot(img_name)
